### PR TITLE
Disable calling _swift_stdlib_dyld_is_objc_constant_string, it's unnecessary currently

### PR DESCRIPTION
**Pull request by Catfish-Man**
Fixes rdar://138093973

This is just dead code currently

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please apple silicon benchmark

File: stdlib/public/core/StringBridge.swift [hunk 1]
```diff
@@ -516,17 +516,8 @@ private func formConstantTaggedCocoaString(
...
-  // Check if what we're pointing to is actually a valid tagged constant
-  guard _swift_stdlib_dyld_is_objc_constant_string(constantPtr) == 1 else {
-    return nil
-  }
-
-  let retaggedPointer = UInt(bitPattern: constantPtr) | expectedConstantTagValue
-
-  return unsafeBitCast(retaggedPointer, to: AnyObject.self)
+// See git history if we ever need the implementation of this again
+  return nil
```
Comment (kperryua) [inline stdlib/public/core/StringBridge.swift:520]: Should we remove `expectedConstantTagValue` as well? Is `constantTagMask` dead as well?

Comment (stephentyrone) [reply stdlib/public/core/StringBridge.swift:520]: Should we eliminate the `#if` conditional entirely?

**Comment by Catfish-Man** [comment]
@swift-ci please test

File: stdlib/public/core/StringBridge.swift (hunk 1 above)
Comment (Catfish-Man) [reply stdlib/public/core/StringBridge.swift]: I updated the diff to remove everything related to this rather than just the slow bit

**Review by kperryua — APPROVED** [review]

**Comment by Catfish-Man** [comment]
@swift-ci please apple silicon benchmark

**Review by glessard — APPROVED** [review]

**Pull request merged**